models/stics/R/met2model.STICS.R

```diff
@@ -0,0 +1,183 @@
+#-------------------------------------------------------------------------------
+# Copyright (c) 2012 University of Illinois, NCSA.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the 
+# University of Illinois/NCSA Open Source License
+# which accompanies this distribution, and is available at
+# http://opensource.ncsa.illinois.edu/license.html
+#-------------------------------------------------------------------------------
+
+##-------------------------------------------------------------------------------------------------#
+##' Converts a met CF file to a model specific met file. The input
+##' files are calld <in.path>/<in.prefix>.YYYY.cf
+##'
+##' @name met2model.STICS
+##' @title Write STICS met files
+##' @param in.path path on disk where CF file lives
+##' @param in.prefix prefix for each file
+##' @param outfolder location where model specific output is written.
+##' @param start_date start date of the simulation
+##' @param end_date end date of the simulation
+##' @param overwrite logical: replace output files if they already exist?
+##' @param ... other arguments passed to function
+##' @return results 
+##' @export
+##' @author Istem Fer
+##-------------------------------------------------------------------------------------------------#
+met2model.STICS <- function(in.path, in.prefix, outfolder, start_date, end_date,
+                            overwrite = FALSE, ...) {
+  
+  PEcAn.logger::logger.info("START met2model.STICS")
+  
+  start_date  <- as.POSIXlt(start_date, tz = "UTC")
+  end_date    <- as.POSIXlt(end_date, tz = "UTC")
+  start_year  <- lubridate::year(start_date)
+  end_year    <- lubridate::year(end_date)
+  
+  # starting with the easiest case, full years
+  # STICS looks for different input files for each year
+  out.files      <- paste(in.prefix, seq(start_year, end_year), "climate", sep = ".")
+  out.files.full <- file.path(outfolder, out.files)
+  
+  results <- data.frame(file = out.files.full,
+                        host = PEcAn.remote::fqdn(),
+                        mimetype = "text/plain",
+                        formatname = "climate",
+                        startdate = start_date,
+                        enddate = end_date,
+                        dbfile.name = out.files,
+                        stringsAsFactors = FALSE)
+  PEcAn.logger::logger.info("internal results")
+  PEcAn.logger::logger.info(results)
+  
+  
+  ## check to see if the outfolder is defined, if not create directory for output
+  if (!file.exists(outfolder)) {
+    dir.create(outfolder)
```

Comment:
Does this want a recursive = TRUE for cases where outfolder contains multiple new levels?

Reply:
not sure what you mean by "outfolder contains multiple new levels". I'm trying to check if  something like "/path/to/model/specific/inputs/" exist, and create if not. "/path/to/model/specific/inputs/" usually don't have multiple levels (as I said, not sure what that means)?

Reply:
I'm thinking of a case where `/path/to/model` exists but doesn't yet contain a `specific/`, so `dir.create("/path/to/model/specific/inputs")` will fail but `dir.create("/path/to/model/specific/inputs", recursive = TRUE)` will succeed